docs/source/design.rst

```diff
@@ -539,6 +539,9 @@ supported systems.
 Guest mode
 ----------
 
+One of Trio's more unusual features is that it supports being run in
+"guest mode" on top of some other event loop (the "host").
```

Comment:
Should we link to the guest mode docs?

Reply:
Oh right, I actually decided to put the high level "how does this even work" stuff into the main docs, and was going to delete this section. But then I forgot :-)

I guess the diagram is probably helpful... I'll move it to a comment by the functions.